src/views/virtualmachines/actions/components/DeleteVMModal/DeleteVMModal.tsx

```diff
@@ -86,6 +91,9 @@ const DeleteVMModal: React.FC<DeleteVMModalProps> = ({ vm, isOpen, onClose }) =>
     await k8sDelete({
       model: VirtualMachineModel,
       resource: updatedVM,
+      json: withGracePeriod
+        ? { kind: 'DeleteOptions', apiVersion: 'v1', gracePeriodSeconds }
+        : null,
```

Comment:
please try seems to be working here withGracePeriod ?? { kind: 'DeleteOptions', apiVersion: 'v1', gracePeriodSeconds }

Reply:
the operator `??` would return deleteoptions only if withGracePeriod is `null` or `undefined`